test/Transforms/multicontrol_decomposition.qke

```diff
@@ -0,0 +1,64 @@
+// ========================================================================== //
+// Copyright (c) 2022 - 2023 NVIDIA Corporation & Affiliates.                 //
+// All rights reserved.                                                       //
+//                                                                            //
+// This source code and the accompanying materials are made available under   //
+// the terms of the Apache License 2.0 which accompanies this distribution.   //
+// ========================================================================== //
+
+// RUN: cudaq-opt -pass-pipeline='builtin.module(func.func(quake-multicontrol-decomposition))' %s | FileCheck %s
+// RUN: cudaq-opt -pass-pipeline='builtin.module(func.func(quake-multicontrol-decomposition))' %s | CircuitCheck %s
+
+// CHECK-LABEL: func.func @ccr1
+func.func @ccr1(%c0: !quake.ref, %c1: !quake.ref, %t: !quake.ref) {
+  %0 = arith.constant 1.57079632679489660 : f64
+  // CHECK-NOT: quake.r1 (%{{.*}})[%{{.*}}, %{{.*}}] %{{.*}}
+  quake.r1 (%0)[%c0, %c1] %t : (f64, !quake.ref, !quake.ref, !quake.ref) -> ()
+  return
+}
+
+// CHECK-LABEL: func.func @cccr1
+func.func @cccr1(%c0: !quake.ref,  %c1: !quake.ref, %c2: !quake.ref, %t: !quake.ref) {
+  %0 = arith.constant 1.57079632679489660 : f64
+  // CHECK-NOT: quake.r1 (%{{.*}})[%{{.*}}, %{{.*}}, %{{.*}}] %{{.*}}
```

Comment:
Can we also check that the created ops are correct?

Reply:
I'm unsure if it is necessary to check for a precise sequence of operations. Here, `FileCheck` makes sure that the operation was indeed decomposed (by checking that it is not present) and `CircuitCheck` makes sure that whatever the decomposition was, it is correct.